### PR TITLE
Address pylint messages

### PR DESCRIPTION
**Pull request by vinodsathe**
Addressed Pylint messages

### CODE REVIEW

File: skabase/SKABaseDevice/SKABaseDevice.py [hunk 1]
```diff
@@ -137,8 +148,8 @@ def get_device_attributes(self, with_value=False,
 
         # Cannot loop over the attr_list object (not python-wrapped): raises TypeError:
         # No to_python (by-value) converter found for C++ type: Tango::Attribute*
-        for index in range(len(attr_list)):
-
+        # for index in range(len(attr_list)):
+        for index in enumerate(attr_list):
```
Comment (ajoubertza) [inline skabase/SKABaseDevice/SKABaseDevice.py]: If you use `enumerate`, `index` will be a tuple, so line 153 is likely to break.

File: skabase/SKACapability/test/SKACapability_test.py [hunk 2]
```diff
@@ -8,16 +8,22 @@
...
 import pytest
 from tango import DevState
-import re
+
+# Path
+path = os.path.join(os.path.dirname(__file__), os.pardir)
+sys.path.insert(0, os.path.abspath(path))
+
+
+
```
Comment (ajoubertza) [inline skabase/SKACapability/test/SKACapability_test.py]: Too many blank lines - doesn't the linter pick this up?

File: skabase/SKALogger/SKALogger.py [hunk 3]
```diff
@@ -11,17 +11,15 @@
...
-from tango.server import run, DeviceMeta, command
-
-# Additional import
 # PROTECTED REGION ID(SKALogger.additionnal_import) ENABLED START #
-from future.utils import with_metaclass
 # standard imports
 import os
 import sys
+from future.utils import with_metaclass
+# tango imports
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: In other files there's a blank line before this comment - please keep it consistent.

File: skabase/SKALogger/SKALogger.py [hunk 4]
```diff
@@ -30,20 +28,14 @@
...
-
-logger_dict = {}
-logging.basicConfig()
-logger = logging.getLogger("SKALogger")
-logger.setLevel(logging.DEBUG)
-syslog = SysLogHandler(address='/var/run/rsyslog/dev/log', facility='syslog')
-formatter = logging.Formatter('%(name)s: %(levelname)s %(module)s %(message)r')
-syslog.setFormatter(formatter)
-logger.addHandler(syslog)
+# logger_dict = {}
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Commenting out large blocks of code isn't recommended.  Also this will likely cause a merge conflict with changes in PR #109. 

More importantly, this PR is about linting, but here is a behavioural change.

File: skabase/SKALogger/SKALogger.py [hunk 5]
```diff
@@ -80,18 +72,18 @@ def init_device(self):
...
-            logger.setLevel(logging.ERROR)
-        elif self._storage_logging_level == int(tango.LogLevel.LOG_WARNING):
-            logger.setLevel(logging.WARNING)
-        elif self._storage_logging_level == int(tango.LogLevel.LOG_INFO):
-            logger.setLevel(logging.INFO)
-        elif self._storage_logging_level == int(tango.LogLevel.LOG_DEBUG):
-            logger.setLevel(logging.DEBUG)
-        else:
-            logger.setLevel(logging.DEBUG)
+        # if self._storage_logging_level == int(tango.LogLevel.LOG_FATAL):
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Where is this all commented out?  Another behavioural change.

File: skabase/SKALogger/test/SKALogger_test.py [hunk 6]
```diff
@@ -8,23 +8,24 @@
...
-# Path
+
+# standard imports
 import sys
 import os
 
-path = os.path.join(os.path.dirname(__file__), os.pardir)
-sys.path.insert(0, os.path.abspath(path))
-
 # Imports
```
Comment (ajoubertza) [inline skabase/SKALogger/test/SKALogger_test.py]: The formatting of this section is not consistent with the other changes in this PR.

File: skabase/SKALogger/test/SKALogger_test.py [hunk 7]
```diff
@@ -85,8 +86,7 @@ def test_Status(self, tango_context):
     @pytest.mark.parametrize("logging_level", [int(tango.LogLevel.LOG_INFO)])
     @pytest.mark.parametrize("logging_target", ["logger/test/1"])
     # PROTECTED REGION END #    //  SKALogger.test_SetCentralLoggingLevel_decorators
-    def test_SetCentralLoggingLevel(self, tango_context, setup_log_test_device,
```
Comment (ajoubertza) [inline skabase/SKALogger/test/SKALogger_test.py]: `setup_log_test_device` might be unused in this function, but it is a test fixture and including it means that `SKATestDevice/SKATestDevice.py` will be launched.  Are we sure that is no longer required?

File: skabase/SKALogger/test/SKALogger_test.py [hunk 8]
```diff
@@ -125,8 +124,7 @@ def test_SetElementLoggingLevel(self, tango_context, setup_log_test_device,
     @pytest.mark.parametrize("logging_level", [int(tango.LogLevel.LOG_WARN)])
     @pytest.mark.parametrize("logging_target", ["logger/test/1"])
     # PROTECTED REGION END #    //  SKALogger.test_SetStorageLoggingLevel_decorators
-    def test_SetStorageLoggingLevel(self, tango_context, setup_log_test_device,
```
Comment (ajoubertza) [inline skabase/SKALogger/test/SKALogger_test.py]: Same comment about fixture being removed.

File: skabase/SKASubarray/SKASubarray.py [hunk 9]
```diff
@@ -10,26 +10,25 @@
...
+from tango import DebugIt
+from tango.server import run, DeviceMeta, attribute, command
+from tango.server import device_property
+from tango import DeviceProxy, Except, ErrSeverity, DevState
+
 # SKA specific imports
 from skabase import release
-
+from SKAObsDevice import SKAObsDevice
 file_path = os.path.dirname(os.path.abspath(__file__))
```
Comment (ajoubertza) [inline skabase/SKASubarray/SKASubarray.py]: Consistency:  no `# Path` comment here, like other files have.

File: skabase/auxiliary/utils.py [hunk 10]
```diff
@@ -94,7 +93,6 @@ def exception_manager(cls, arguments="", callback=None):
         # cls.exception(command_name=class_name + "::" + calframe[2][3],
         #               command_inputs=str(arguments),
```
Comment (ajoubertza) [inline skabase/auxiliary/utils.py]: Now that the `arguments` parameter has been removed, we might as well remove this whole commented out block.

File: skabase/conftest.py [hunk 11]
```diff
@@ -2,11 +2,14 @@
...
-import pytest
-import importlib
 import os
 import time
+import importlib
+import mock
+import pytest
+
+
+
```
Comment (ajoubertza) [inline skabase/conftest.py]: Too many blank lines?

File: skabase/SKAAlarmHandler/SKAAlarmHandler.py [hunk 12]
```diff
@@ -11,18 +11,16 @@
...
 #standard import
 import os
 import sys
 from future.utils import with_metaclass
 
+# tango imports
+from tango import DebugIt
+from tango.server import run, DeviceMeta, attribute, command, device_property
+
 # SKA specific imports\
```
Comment (ajoubertza) [inline skabase/SKAAlarmHandler/SKAAlarmHandler.py]: Remove trailing slash.

File: skabase/SKAAlarmHandler/SKAAlarmHandler.py [hunk 13]
```diff
@@ -11,18 +11,16 @@
...
 attribute values and quality factors, and are separate from the "built-in" TANGO attribute alarms.
 """
-
-# tango imports
-from tango import DebugIt
-from tango.server import run, DeviceMeta, attribute, command, device_property
-
-# Additional import
 # PROTECTED REGION ID(SKAAlarmHandler.additionnal_import) ENABLED START #
 #standard import
```
Comment (ajoubertza) [inline skabase/SKAAlarmHandler/SKAAlarmHandler.py]: Space after `#`.
`imports` rather than `import`.

File: skabase/SKABaseDevice/SKABaseDevice.py [hunk 14]
```diff
@@ -8,34 +8,37 @@
...
 """
+
+# PROTECTED REGION ID(SKABaseDevice.additionnal_import) ENABLED START #
+# standard imports
+import os
+import sys
+import logging
+import logging.handlers
+from logging.handlers import SysLogHandler
+import json
```
Comment (ajoubertza) [inline skabase/SKABaseDevice/SKABaseDevice.py]: Pep8 says line like `import foo` should be before lines like `from bar import baz`.

File: skabase/SKABaseDevice/SKABaseDevice.py [hunk 15]
```diff
@@ -137,8 +148,8 @@ def get_device_attributes(self, with_value=False,
 
         # Cannot loop over the attr_list object (not python-wrapped): raises TypeError:
         # No to_python (by-value) converter found for C++ type: Tango::Attribute*
-        for index in range(len(attr_list)):
-
+        # for index in range(len(attr_list)):
```
Comment (ajoubertza) [inline skabase/SKABaseDevice/SKABaseDevice.py]: Commented out code is not useful.

**Review by ajoubertza — CHANGES_REQUESTED** [review]
Thanks for fixing so many files.  Two main issues:
- avoid behavioural changes
- be consistent. e.g.

Is it `#standard import` or `# standard import` or `# Standard import` or `# Standard imports`?
Note spacing, capitalisation, plural or singular.

Is it always "Standard imports", then "Imports", then "Tango imports"?  Sometimes they are not all there.

File: skabase/SKAAlarmHandler/SKAAlarmHandler.py (hunk 13 above)
Comment (vinodsathe) [reply skabase/SKAAlarmHandler/SKAAlarmHandler.py]: Fixed

Comment (vinodsathe) [reply skabase/SKAAlarmHandler/SKAAlarmHandler.py]: Checked and fixed in all other files.

File: skabase/conftest.py (hunk 11 above)
Comment (vinodsathe) [reply skabase/conftest.py]: Checked and fixed in all other files as well

File: skabase/SKACapability/test/SKACapability_test.py (hunk 2 above)
Comment (vinodsathe) [reply skabase/SKACapability/test/SKACapability_test.py]: Checked and fixed in all other files.

Comment (vinodsathe) [reply skabase/SKACapability/test/SKACapability_test.py]: Wherever it picks up it is being fixed.

File: skabase/SKALogger/SKALogger.py (hunk 3 above)
Comment (vinodsathe) [reply skabase/SKALogger/SKALogger.py]: Checked and fixed in all other files.

File: skabase/SKALogger/SKALogger.py (hunk 5 above)
Comment (vinodsathe) [reply skabase/SKALogger/SKALogger.py]: Reverted back the changes. This is addressed in other pull request.

File: skabase/SKALogger/SKALogger.py (hunk 4 above)
Comment (vinodsathe) [reply skabase/SKALogger/SKALogger.py]: Reverted back the changes. This is addressed in other pull request.

File: skabase/SKABaseDevice/SKABaseDevice.py (hunk 14 above)
Comment (vinodsathe) [reply skabase/SKABaseDevice/SKABaseDevice.py]: Addressed

File: skabase/SKAAlarmHandler/SKAAlarmHandler.py [hunk 16]
```diff
@@ -4,31 +4,29 @@
...
-# SKA specific imports\
+# SKA specific imports
 from skabase import release
-
 file_path = os.path.dirname(os.path.abspath(__file__))
 basedevice_path = os.path.abspath(os.path.join(file_path, os.pardir)) + "/SKABaseDevice"
 sys.path.insert(0, basedevice_path)
 from SKABaseDevice import SKABaseDevice
-# PROTECTED REGION END #    //  SKAAlarmHandler.additionnal_import
+# PROTECTED REGION END #    //  SKAAlarmHandler.additionnal_imports
```
Comment (ajoubertza) [inline skabase/SKAAlarmHandler/SKAAlarmHandler.py]: This should not have changed - see line 13.  
(These lines are used by POGO, but since its usage seems to be abandoned, won't make much difference).

File: skabase/conftest.py [hunk 17]
```diff
@@ -83,7 +83,7 @@ def setup_log_test_device():
     # run test device
     file_path = os.path.dirname(os.path.abspath(__file__))
     testdevice_path = os.path.abspath(os.path.join(file_path, os.curdir)) + "/SKATestDevice/SKATestDevice.py"
-    cmdline = 'python3 ' + testdevice_path + ' ' + '1 &'
+    cmdline = 'python3 ' + testdevice_path + ' ' + '01 &'
```
Comment (ajoubertza) [inline skabase/conftest.py]: Why this change?  Commit message just says what changed, but not why.

File: skabase/SKATestDevice/SKATestDevice.py [hunk 18]
```diff
@@ -25,15 +23,15 @@
 
 # SKA specific imports
 from skabase import release
-from skabase.auxiliary.utils import (exception_manager, convert_api_value, coerce_value)
-from SKABaseDevice import SKABaseDevice
-
 
 file_path = os.path.dirname(os.path.abspath(__file__))
 basedevice_path = os.path.abspath(os.path.join(file_path, os.pardir)) + "/SKABaseDevice"
 sys.path.insert(0, basedevice_path)
```
Comment (ajoubertza) [inline skabase/SKATestDevice/SKATestDevice.py]: Not something to fix now, but it is pretty weird that all the devices modules have to modify the path to get imports to work. It suggests something is not quite right in how the packages are set up.

**Review by ajoubertza — APPROVED** [review]

File: skabase/conftest.py (hunk 17 above)
Comment (adityadangeska) [reply skabase/conftest.py]: The instance name of the device server which is getting registered in the script is 01. Locally we already had a device server named 1 but in the the pipeline, the execution failed. So we had to change it.
Strangely in earlier pipeline runs this error did not occur as this code was added long back.

File: skabase/SKATestDevice/SKATestDevice.py (hunk 18 above)
Comment (adityadangeska) [reply skabase/SKATestDevice/SKATestDevice.py]: I request you to go through the packaging and code structure and let us know your inputs on what should be the best way to package. We will be happy to improve the packaging.

File: skabase/SKAAlarmHandler/SKAAlarmHandler.py (hunk 16 above)
Comment (adityadangeska) [reply skabase/SKAAlarmHandler/SKAAlarmHandler.py]: Ok. That was a typo. Will revert the change.

File: skabase/SKALogger/test/SKALogger_test.py (hunk 7 above)
Comment (vinodsathe) [reply skabase/SKALogger/test/SKALogger_test.py]: We are instantiating separate container for SKATestDevice. However, for testing in the local environment, we still need this fixture. We had removed this in anticipation that we will do not more testing in the local environment. But as a second thought, we still may need to run test cases locally. So we will revert this change for some time and if not found useful, will remove it.

File: skabase/SKALogger/test/SKALogger_test.py (hunk 8 above)
Comment (vinodsathe) [reply skabase/SKALogger/test/SKALogger_test.py]: We are instantiating separate container for SKATestDevice. However, for testing in the local environment, we still need this fixture. We had removed this in anticipation that we will do not more testing in the local environment. But as a second thought, we still may need to run test cases locally. So we will revert this change for some time and if not found useful, will remove it.

File: skabase/SKALogger/test/SKALogger_test.py (hunk 6 above)
Comment (vinodsathe) [reply skabase/SKALogger/test/SKALogger_test.py]: It looks consistent all across now
